### PR TITLE
fix: handle savers hard cap reached

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR ensures we handle the pseudo-halted state of savers hard cap reached, i.e a holistic hard cap across savers which effectively makes all deposits halted.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/4752
- closes https://github.com/shapeshift/web/issues/4766

## Risk

None

## Testing

- Ensure "Deposit halted" shows in overview, and the deposit button is hidden while the hard cap is reached (which is the case at the time this is authored)

### Engineering

- ☝🏽 
### Operations

- ☝🏽 

## Screenshots (if applicable)

- Develop

Screenshot 2023-06-15 at 21 50 11

- This diff

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4767** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4767" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4767?utm_source=stack-comment).

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx [hunk 1]
```diff
@@ -285,23 +307,27 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
...
-              as={Link}
-              href={`https://twitter.com/intent/tweet?text=Hey%20%40THORChain%20%23raisethecaps%20already%20so%20I%20can%20deposit%20%23${underlyingAsset?.symbol}%20into%20a%20savers%20vault%20at%20%40ShapeShift`}
-              isExternal
-              mt={4}
-              colorScheme='twitter'
-              rightIcon={<FaTwitter />}
-            >
-              @THORChain
-            </Button>
+            {!isHardCapReached && (
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:314]: We don't want to show the "bug THORChain on Twitter" description and button here - this isn't about raising the caps, this is a hollistic max. across the network

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx [hunk 2]
```diff
@@ -85,6 +90,22 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
   const assets = useAppSelector(selectAssets)
   const asset = useAppSelector(state => selectAssetById(state, assetId))
 
+  useEffect(() => {
+    ;(async () => {
+      if (!(asset && assetId)) return
+
+      const maybeQuote = await getMaybeThorchainSaversDepositQuote({
```
Comment (gomesalexandre) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:97]: Tries to get a quote for the current asset's minimum

File: src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx [hunk 3]
```diff
@@ -85,6 +90,22 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
...
+        asset,
+        amountCryptoBaseUnit: THORCHAIN_SAVERS_DUST_THRESHOLDS[assetId],
+      })
+      if (
+        maybeQuote.isErr() &&
+        maybeQuote.unwrapErr().includes('add liquidity rune is more than bond')
```
Comment (kaladinlight) [inline src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:103]: do they have a custom error code out of curiosity for this hard cap case? only concern is I feel like error text has higher chance to change than an error code

Comment (gomesalexandre) [reply src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx:103]: Unfortunately not, if we hit `/thorchain/quote/saver/deposit?asset=BTC.BTC&amount=10000000`, we get a GraphQL style 200, no HTTP error code, no `code` property, only a message
image

**Pull request merged**